subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java

```diff
@@ -50,67 +44,41 @@
 import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.plugins.BasePlugin;
 import org.gradle.api.plugins.JavaBasePlugin;
-import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.plugins.JavaPluginExtension;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskContainer;
-import org.gradle.api.tasks.TaskDependency;
 import org.gradle.api.tasks.TaskProvider;
 import org.gradle.api.tasks.bundling.Jar;
 import org.gradle.api.tasks.compile.AbstractCompile;
 import org.gradle.api.tasks.compile.CompileOptions;
 import org.gradle.api.tasks.compile.JavaCompile;
 import org.gradle.api.tasks.javadoc.Javadoc;
 import org.gradle.internal.Cast;
-import org.gradle.internal.Factory;
 import org.gradle.internal.jvm.JavaModuleDetector;
 import org.gradle.language.base.plugins.LifecycleBasePlugin;
 import org.gradle.util.TextUtil;
 
 import javax.annotation.Nullable;
-import java.io.File;
 import java.util.Date;
 import java.util.List;
-import java.util.Set;
 import java.util.concurrent.Callable;
-
-import static org.gradle.api.attributes.Bundling.BUNDLING_ATTRIBUTE;
-import static org.gradle.api.attributes.Usage.USAGE_ATTRIBUTE;
+import java.util.function.Function;
 
 /**
  * Helpers for Jvm plugins. They are in a separate class so that they don't leak
  * into the public API.
  */
 public class JvmPluginsHelper {
```

Comment:
Looks like there is more in this class that can be replaced by calls to the service now (?)

E.g. parts of `addApiToSourceSet`, `configureDocumentationVariantWithArtifact`, ...

Reply:
Well in the end we should get rid of this. I didn't want to introduce any circular dependency (service -> helper -> service).

Reply:
I guess then it is the other way around. The places that call these methods should use the service instead of the helper. But we can continue work on this later.